DOMParser: Tweak tests for interface of returned document

emilio opened the pull request:
See https://github.com/whatwg/html/issues/11372. This matches all browsers.

emilio committed:
DOMParser: Tweak tests for interface of returned document

See https://github.com/whatwg/html/issues/11372. This matches all
browsers.

domenic reviewed (commented):
Given that these interfaces don't exist in specifications, I'm unsure if we should test them in this way, or just test `instanceof Document`.

I guess we could also add `tentative` tests for these specific instance cases.

annevk commented:
Yeah, as-is this would require specification changes. Which we should probably make, but figuring out how to solve this was somewhat non-trivial I believe.

emilio commented:
@domenic @annevk Ok I just removed those tests for now. Those tests would also not be correct if we removed the `XMLDocument` interface (because they would do `instanceof undefined`).

emilio commented:
(`instanceof Document` is tested in `checkMetadata`)

annevk commented:
Per https://dom.spec.whatwg.org/#xmldocument we do define that `XMLDocument` exists. So we should go with the original approach and update the specification to match.
